internal/services/appconfiguration/client/client.go

```diff
@@ -1,12 +1,49 @@
 package client
 
 import (
+	"context"
+	"fmt"
+
 	"github.com/hashicorp/terraform-provider-azurerm/internal/common"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration/sdk/2020-06-01/configurationstores"
+
+	"github.com/Azure/go-autorest/autorest"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration/sdk/1.0/appconfiguration"
 )
 
 type Client struct {
 	ConfigurationStoresClient *configurationstores.ConfigurationStoresClient
+	tokenFunc                 func(endpoint string) (autorest.Authorizer, error)
+	configureClientFunc       func(c *autorest.Client, authorizer autorest.Authorizer)
+}
+
+func (c Client) DataPlaneClient(ctx context.Context, configurationStoreId string) (*appconfiguration.BaseClient, error) {
+	appConfigId, err := configurationstores.ParseConfigurationStoreID(configurationStoreId)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: caching all of this
+	appConfig, err := c.ConfigurationStoresClient.Get(ctx, *appConfigId)
+	if err != nil {
+		// TODO: if not found etc
+		return nil, err
+	}
+
+	if appConfig.Model == nil || appConfig.Model.Properties == nil || appConfig.Model.Properties.Endpoint == nil {
+		return nil, fmt.Errorf("endpoint was nil")
+	}
+
+	appConfigAuth, err := c.tokenFunc(*appConfig.Model.Properties.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	endpoint := *appConfig.Model.Properties.Endpoint
```

Comment:
_minor_ but we may as well move this up to :37 and use it there, rather than de-refing this twice?